src/sardana/spock/magic.py

```diff
@@ -97,6 +106,13 @@ def showscan(self, parameter_s=''):
         print("Qt binding is not available. Showscan cannot work without it.")
         return
 
+    try:
+        if not xsession_available():
+            raise RuntimeError
+    except RuntimeError:
+        print("No X-session available. Showscan cannot work without it.")
```

Comment:
Does everyone know what X-session means? Better to say "No display available."?

Reply:
You are correct. I think 99.99% of the users have no idea what that means. But stating "No display available" might also confuse them since they are staring at one :-)

Reply:
Yes you are right, "No display" isn't much better.. 
How about something like "Running in a text-only terminal. Showscan cannot work without a graphical display."?

Reply:
I think this is the kind of message the user would appreciate: A warning about "missing features"

Reply:
At DESY we don't use `Showscan` so the above message would be confusing. I would prefer general one about graphical display

Reply:
I will change it to something along these lines: "Running without graphical user interface support". Fine?

Reply:
> At DESY we don't use `Showscan` so the above message would be confusing. I would prefer general one about graphical display

I agree that instead of showscan should be written something more general, but only clarify that in DESY showscan
is indeed used in some beamlines.

Reply:
> I will change it to something along these lines: "Running without graphical user interface support". Fine?

In my opinion, yes

Reply:
I'm also fine with that.